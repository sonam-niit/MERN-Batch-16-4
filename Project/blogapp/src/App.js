import { BrowserRouter, Route, Routes } from 'react-router-dom';
import './App.css';
import Header from './components/header';
import PostList from './components/PostList';
import Profile from './components/profile';
import Comments from './components/Comments';

function App() {
  return (
    <BrowserRouter>
      <div className="container-fluid">
        <Header />
      </div>

      <Routes>
        <Route path='/' element={<PostList />}></Route>
        <Route path='/profile/:id' element={<Profile />}></Route>
        <Route path='/post/:id' element={<Comments />}></Route>
      </Routes>
    </BrowserRouter>
  );
}

export default App;
