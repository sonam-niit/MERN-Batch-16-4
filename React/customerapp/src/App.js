import { BrowserRouter, Navigate, Route, Routes } from "react-router-dom";
import About from "./components/about";
import ContactUs from "./components/contact";
import CustomerManagement from "./components/custManage";
import Navbar from "./components/navbar";
import Home from "./components/home";
import List from "./components/list";
import Details from "./components/details";
import AsyncStateExample from "./components/asyncstate";

function App() {

  return (
    <BrowserRouter>
      <div className='container'>
        <h1>Online Shopping Portal</h1>
        <Navbar />

        <Routes>
          <Route path="/" element={<Navigate to="/home" replace/>} ></Route>
          {/* Replace keep your history clean, helps you to avoid extra redirects after click on back link */}
          <Route path="/home" element={<Home />}></Route>
          <Route path="/about" element={<About />}></Route>
          <Route path="/contact" element={<ContactUs />}></Route>
          <Route path="/custmanage" element={<CustomerManagement />}></Route>
          <Route path="/list" element={<List />}></Route>
          <Route path="/details/:id" element={<Details />}></Route>
          <Route path="/async" element={<AsyncStateExample />}></Route>
          {/* Dynamic Route */}
        </Routes>
      </div>
    </BrowserRouter>)
}

export default App;
