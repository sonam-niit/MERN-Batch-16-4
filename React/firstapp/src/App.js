import Session1 from "./components/session1";
import Session2 from "./components/session2";

function App() {

  return (
    <div>
      <Session1 />
      <Session2 />
    </div>
  );
}

export default App;
