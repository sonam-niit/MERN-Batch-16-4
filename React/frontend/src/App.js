import { BrowserRouter, Route, Routes } from "react-router-dom";
import Navigation from "./components/navbar";
import UserList from "./components/userlist";
import AddUser from "./components/adduser";
import EditUser from "./components/edituser";

function App() {
  return (
    <BrowserRouter>
      <div className="container">
        <Navigation />

      <Routes>
        <Route path="/list" element={<UserList />}></Route>
        <Route path="/add" element={<AddUser />}></Route>
        <Route path="/edit/:id" element={<EditUser />}></Route>
      </Routes>
      </div>
    </BrowserRouter>
  );
}

export default App;
