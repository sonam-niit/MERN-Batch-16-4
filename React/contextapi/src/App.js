import logo from './logo.svg';
import './App.css';
import GrandParent from './components/usecase1/GrandParent';
import MyGrandParent from './components/usecase2/MyGrandParent';
import { ThemeProvider } from './components/usecase3/TheameContext';
import ThemedComponent from './components/usecase3/ThemedComponent';
import { AuthProvide } from './components/usecase4/AuthContext';
import UserInfo from './components/usecase4/UserInfo';

function App() {
  return (
    <AuthProvide>
      <h1>Authentication App</h1>
      <UserInfo />
    </AuthProvide>
    // <div className="App">
    //   {/* Prop Drilling */}
    //   {/* <GrandParent /> */}

    //   {/* Using Context API */}
    //   {/* <MyGrandParent /> */}
    //   {/* Use Thme Context */}
    //   <ThemeProvider>
    //     <h1>Themed App</h1>
    //     <ThemedComponent />
    //   </ThemeProvider>
    // </div>
  );
}

export default App;
