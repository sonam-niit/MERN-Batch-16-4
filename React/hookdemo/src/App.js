import logo from './logo.svg';
import './App.css';
import UseEffectDemo1 from './components/useeffect1';
import Timer from './components/timer';
import { BrowserRouter, Link, Route, Routes } from 'react-router-dom';
import useDocumentTitle from './components/useDocumentTitle';
import DynamicDocumentTitle from './components/useDocumentTitle';
import UseCaseInputValidation from './components/UseCaseInputValidation';
import UseCaseLiveFilter from './components/UseCaseLiveFilter';
import RealTimeClock from './components/realtimeclock';

function Home() {
  // useDocumentTitle('Home Page')
  return (<div>
            <DynamicDocumentTitle title="Home Page"/>
            <h1>Home Component</h1>
          </div>)
}
function About() {
  //useDocumentTitle('About Us')
  return (<div>
    <DynamicDocumentTitle title="About Us"/>
    <h1>About Component</h1>
  </div>)
}
function Contact() {
  //useDocumentTitle('Contact Us')
  return (<div>
    <DynamicDocumentTitle title="Contact Us"/>
    <h1>Contact Component</h1>
  </div>)
}
function App() {
  return (
    <BrowserRouter>
      <RealTimeClock />
      <UseCaseInputValidation />
      <UseCaseLiveFilter />
      <Link to="/home">Home</Link>
      <Link to="/about">About</Link>
      <Link to="/contact">Contact</Link>
      <Routes>
        <Route path='/home' element={<Home />}></Route>
        <Route path='/about' element={<About />}></Route>
        <Route path='/contact' element={<Contact />}></Route>
      </Routes>
    </BrowserRouter>
  );
}


export default App;
