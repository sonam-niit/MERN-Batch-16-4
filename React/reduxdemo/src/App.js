import './App.css';
import UseCase1 from './usecase1/useCase1';
import UseCase2 from './usecase1/useCase2';

function App() {
  return (
    <div className="App">
        <UseCase1 />
        <UseCase2 />
    </div>
  );
}

export default App;
