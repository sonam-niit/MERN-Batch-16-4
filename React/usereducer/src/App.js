import logo from './logo.svg';
import './App.css';
import Counter from './usecase1/Counter';
import Bugs from './usecase2/bugs';
import SimpleForm from './usecase3/SimpleForm';
import ShoppingCart from './usecase4/ShoppingCart';

function App() {
  return (
    <div className="App">
      {/* <Counter />
      <Bugs />
      <SimpleForm /> */}
      <ShoppingCart />
    </div>
  );
}

export default App;
