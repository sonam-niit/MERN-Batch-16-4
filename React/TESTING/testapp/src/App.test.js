import { cleanup, render, screen } from '@testing-library/react';
import App from './App';

//SnapShot
afterEach(cleanup);

it('should take a snapshot',()=>{
  const {asFragment}= render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
})

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
