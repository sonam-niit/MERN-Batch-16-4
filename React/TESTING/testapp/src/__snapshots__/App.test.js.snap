// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      learn react
    </h1>
    <button>
      Click Me
    </button>
    <h2>
      Sonam Soni
    </h2>
  </div>
</DocumentFragment>
`;
